WebAPI/LearningHub.Nhs.Database/Stored Procedures/Hierarchy/CheckCatalogueHasExternalReference.sql

```diff
@@ -0,0 +1,105 @@
+﻿-------------------------------------------------------------------------------
+-- Author       Sarathlal
+-- Created      29 August 2024
+-- Purpose      Return if Node has an external reference
+--
+-- Modification History
+--
+-- 29-08-2024  SS	Initial Revision.
+
+-------------------------------------------------------------------------------
+CREATE PROCEDURE [hierarchy].[CheckCatalogueHasExternalReference]
+(
+	@CatalogueNodeId int,
+	@HasExternalCatalogueReference bit OUTPUT  
+)
+AS
+
+BEGIN
+		;WITH
+		  cteNode(NodeId,  ParentNodeId, PrimaryCatalogueNodeId, InitialNodePath)
+		  AS
+		  (
+			SELECT 
+				n.Id AS NodeId,
+				ParentNodeId = NULL,
+                nv.PrimaryCatalogueNodeId,
+				CAST(n.Id AS nvarchar(128)) AS InitialNodePath
+			 FROM 
+                 hierarchy.NodePath np
+             INNER JOIN
+			 	hierarchy.[Node] n ON np.NodeId = n.Id
+             INNER JOIN
+                hierarchy.NodeVersion nv ON n.CurrentNodeVersionId = nv.Id
+			 WHERE 
+			 	nv.NodeId=@catalogueNodeId
+                AND nv.VersionStatusId = 2 -- Published
+                AND np.Deleted = 0
+                AND n.Deleted = 0
+                AND nv.Deleted = 0
+
+			UNION ALL
+
+			SELECT 
+				ChildNodeId AS NodeId,
+				nl.ParentNodeId,
+                nv.PrimaryCatalogueNodeId,
+				CAST(cte.InitialNodePath + '\' + CAST(ChildNodeId AS nvarchar(8)) AS nvarchar(128)) AS InitialNodePath
+			FROM 
+				hierarchy.NodeLink nl
+			INNER JOIN
+				hierarchy.[Node] n ON nl.ChildNodeId = n.Id
+			INNER JOIN	
+				cteNode cte ON nl.ParentNodeId = cte.NodeId
+            INNER JOIN
+                hierarchy.NodeVersion nv ON n.CurrentNodeVersionId = nv.Id
+			WHERE 
+				n.CurrentNodeVersionId IS NOT NULL
+                AND nv.VersionStatusId = 2 -- Published
+                AND n.Deleted = 0
+                AND nl.Deleted = 0
+
+			),
+		
+		 cteResource
+		 AS (
+				SELECT cte.PrimaryCatalogueNodeId 
+							FROM
+								cteNode cte
+				UNION 
+
+				SELECT 
+					rv.PrimaryCatalogueNodeId AS PrimaryCatalogueNodeId
+				FROM
+					hierarchy.NodePath np
+				INNER JOIN
+					hierarchy.NodeResource nr ON np.NodeId = nr.NodeId
+				INNER JOIN
+					resources.[Resource] r ON nr.ResourceId = r.Id
+				INNER JOIN
+					resources.ResourceVersion rv ON rv.resourceId = nr.ResourceId
+				LEFT JOIN
```

Comment:
I can not see any references to these joined tables:
resources.ResourceReference rr
resources.VideoResourceVersion vrv
resources.AudioResourceVersion arv
resources.ResourceVersionEvent rve
resources.ResourceVersion rvd

If they are not used they should be removed